hedera-node/src/main/java/com/hedera/services/grpc/marshalling/BalanceChangeManager.java

```diff
@@ -37,6 +39,7 @@
 	private int levelNo = 0;
 	private int levelStart = 0;
 	private int levelEnd;
+	private Set<Pair<Id, Id>> royaltiesPaid = null;
```

Comment:
should initialize this `royaltiesPaid` to `new HashSet<>()` and remove the `if` on lines 56-58.

Reply:
Creating this unconditionally would affect *every* `CryptoTransfer`, even xfers that will never touch NFTs.